include/wil/resource.h

```diff
@@ -5241,13 +5241,19 @@ namespace wil
     {
         unique_hglobal_locked() = delete;
 
-        unique_hglobal_locked(_In_ STGMEDIUM& medium) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(medium.hGlobal)
+        unique_hglobal_locked(HGLOBAL global) : unique_any<void*, decltype(&::GlobalUnlock), ::GlobalUnlock>(global)
         {
             // GlobalLock returns a pointer to the associated global memory block and that's what callers care about.
-            m_globalMemory = GlobalLock(medium.hGlobal);
+            m_globalMemory = GlobalLock(global);

```

Comment:
If this GlobalLock fails, the global should not be unlocked.

Reply:
note, failure only occurs on invalid handle. the alloc step ensures that the resources are available.

Reply:
OK, I have added call to release() in case of GlobalLock returns NULL.